test/e2e/bcd_consumer_integration_test.go

```diff
@@ -367,34 +357,156 @@ func (s *BCDConsumerIntegrationTestSuite) Test6BabylonFPCascadedSlashing() {
+	s.NoError(err)
+	s.NotNil(txResp)
+
+	// consumer finality provider submits finality signature
+	txResp, err = s.cosmwasmController.SubmitFinalitySig(
+		czFpBTCSK,
+		czFpBTCPK,
+		randListInfo.SRList[0],
+		&randListInfo.PRList[0],
+		randListInfo.ProofList[0].ToProto(),
+		czlatestBlockHeight,
+	)
+	s.NoError(err)
+	s.NotNil(txResp)
+
+	// ensure consumer finality provider's finality signature is received and stored in the smart contract
+	s.Eventually(func() bool {
+		fpSigsResponse, err := s.cosmwasmController.QueryFinalitySignature(consumerFp.BtcPk.MarshalHex(), uint64(czlatestBlockHeight))
+		if err != nil {
+			s.T().Logf("failed to query finality signature: %s", err.Error())
+			return false
+		}
+		if fpSigsResponse == nil || fpSigsResponse.Signature == nil || len(fpSigsResponse.Signature) == 0 {
 			return false
 		}
+		return true
+	}, time.Minute, time.Second*5)
+
+	// consumer finality provider submits invalid finality signature
+	txResp, err = s.cosmwasmController.SubmitInvalidFinalitySig(
+		r,
+		czFpBTCSK,
+		czFpBTCPK,
+		randListInfo.SRList[0],
+		&randListInfo.PRList[0],
+		randListInfo.ProofList[0].ToProto(),
+		czlatestBlockHeight,
+	)
+	s.NoError(err)
+	s.NotNil(txResp)
 
-		// create a map of BTC public keys to ConsumerFpInfoResponse
-		fpMap := make(map[string]cosmwasm.ConsumerFpInfoResponse)
-		for _, fp := range fpsByPower.Fps {
-			fpMap[fp.BtcPkHex] = fp
+	// ensure consumer finality provider is slashed
+	s.Eventually(func() bool {
+		fp, err := s.cosmwasmController.QueryFinalityProvider(consumerFp.BtcPk.MarshalHex())
+		return err == nil && fp != nil && fp.SlashedHeight > 0
+	}, time.Minute, time.Second*5)
+
+	// query and assert consumer finality provider's voting power is zero after slashing
+	s.Eventually(func() bool {
+		fpInfo, err := s.cosmwasmController.QueryFinalityProviderInfo(consumerFp.BtcPk.MustToBTCPK())
+		if err != nil {
+			s.T().Logf("Error querying finality providers by power: %v", err)
+			return false
 		}
 
-		// check if the consumerFp's BTC public key exists in the map
-		consumerFpBtcPkHex := consumerFp.BtcPk.MarshalHex()
-		fpInfo, exists := fpMap[consumerFpBtcPkHex]
-		if !exists {
+		return fpInfo != nil && fpInfo.Power == 0 && fpInfo.BtcPkHex == consumerFp.BtcPk.MarshalHex()
+	}, time.Minute, time.Second*5)
+
+	// check the babylon finality provider's voting power is discounted (cascaded slashing)
+	babylonFpBIP340PKHex := bbntypes.NewBIP340PubKeyFromBTCPK(babylonFpBTCPK2).MarshalHex()
+	s.Eventually(func() bool {
+		fp, err := s.babylonController.QueryFinalityProvider(babylonFpBIP340PKHex)
+		if err != nil {
+			s.T().Logf("Error querying finality provider: %v", err)
 			return false
 		}
+		return fp != nil &&
+			fp.FinalityProvider.VotingPower == 0 && // as there is only 1 delegation which got slashed
+			fp.FinalityProvider.SlashedBtcHeight == 0 // should not be slashed
```

Comment:
👍🏼 Only missing piece here that would be a nice to have, is to have more than one delegation. So that we can check the voting power discount is only for the slashed dels.